testing-tools/src/order_generator/arbitrage.rs

```diff
@@ -0,0 +1,308 @@
+             {:.2} USDC",
+            binance_trade_type,
+            if binance_trade_type == "SELL" { "BUY" } else { "SELL" },
+            binance_price,
+            uniswap_fill_price,
+            binance_amount,
+            uniswap_amount,
+            profit,
+        );
+
+        let zero_for_one = binance_trade_type == "SELL";
+        Some(Self::create_order(zero_for_one, uniswap_amount))
+    }
+
+    fn create_order(zero_for_one: bool, amount: I256) -> TopOfBlockOrder {
+        TopOfBlockOrder {
+            quantity_in: if zero_for_one { amount.abs().try_into().unwrap() } else { 0u128 },
+            quantity_out: if zero_for_one { 0u128 } else { amount.abs().try_into().unwrap() },
+            ..TopOfBlockOrder::default()
+        }
+    }
+
+    fn try_sell_on_uniswap(
+        &self,
+        pool: &EnhancedUniswapPool<Loader>,
+        best_ask: &PriceLevel
+    ) -> (f64, f64, f64, I256) {
+        let eth = pool.token_b;
+        let ask_amount = Decimal::from_f64(best_ask.quantity).unwrap()
+            * Decimal::from(10i64).powu(pool.token_b_decimals.into());
+        let ask_amount_in =
+            I256::from_dec_str(ask_amount.to_string().split('.').next().unwrap()).unwrap();
+
+        let (ask_swap_amount_in, ask_swap_amount_out) =
+            pool.simulate_swap(eth, ask_amount_in, None).unwrap();
+
+        let ask_uniswap_fill_price =
+            self.calculate_uniswap_fill_price(pool, ask_swap_amount_in, ask_swap_amount_out);
+
+        let token_b_scale = Decimal::from(10i64).powu(pool.token_b_decimals.into());
+        let ask_binance_amount = best_ask.quantity;
+        let ask_uniswap_amount =
+            Decimal::from(u128::try_from(ask_swap_amount_out.abs()).unwrap()) / token_b_scale;
+
+        let ask_profit = (ask_uniswap_fill_price.to_f64().unwrap() * ask_binance_amount)
+            - (best_ask.price * ask_uniswap_amount.to_f64().unwrap());
+
+        (ask_profit, ask_binance_amount, ask_uniswap_fill_price.to_f64().unwrap(), ask_amount_in)
+    }
+
+    fn try_buy_on_uniswap(
+        &self,
+        pool: &EnhancedUniswapPool<Loader>,
+        best_bid: &PriceLevel
+    ) -> (f64, f64, f64, I256) {
+        let bid_amount = Decimal::from_f64(best_bid.quantity).unwrap()
+            * Decimal::from_f64(best_bid.price).unwrap()
+            * Decimal::from(10i64).powu(pool.token_a_decimals.into());
+        let bid_amount_in =
+            I256::from_dec_str(bid_amount.to_string().split('.').next().unwrap()).unwrap();
+        let usdc = pool.token_a;
+
+        let (bid_swap_amount_in, bid_swap_amount_out) =
+            pool.simulate_swap(usdc, bid_amount_in, None).unwrap();
+
+        let bid_uniswap_fill_price =
+            self.calculate_uniswap_fill_price(pool, bid_swap_amount_in, bid_swap_amount_out);
+
+        let token_b_scale = Decimal::from(10i64).powu(pool.token_b_decimals.into());
+        let bid_binance_amount = bid_amount / token_b_scale;
```

Comment:
bid amount is adjusted from binance using token_a_decimals. then is divided by token_b_scale to get back to the original bid_binance_amount?